experimental/FTheoryTools/src/AbstractFTheoryModels/methods.jl

```diff
@@ -0,0 +1,538 @@
+
+Resolve an F-theory model by blowing up a locus in the ambient space.
+
+# Examples
+```jldoctest
+julia> B3 = projective_space(NormalToricVariety, 3)
+Normal toric variety
+
+julia> w = torusinvariant_prime_divisors(B3)[1]
+Torus-invariant, prime divisor on a normal toric variety
+
+julia> t = literature_model(arxiv_id = "1109.3454", equation = "3.1", base_space = B3, model_sections = Dict("w" => w), completeness_check = false)
+Construction over concrete base may lead to singularity enhancement. Consider computing singular_loci. However, this may take time!
+
+Global Tate model over a concrete base -- SU(5)xU(1) restricted Tate model based on arXiv paper 1109.3454 Eq. (3.1)
+
+julia> blow_up(t, ["x", "y", "x1"]; coordinate_name = "e1")
+Partially resolved global Tate model over a concrete base -- SU(5)xU(1) restricted Tate model based on arXiv paper 1109.3454 Eq. (3.1)
+```
+Here is an example for a Weierstrass model.
+
+# Examples
+```jldoctest
+julia> B2 = projective_space(NormalToricVariety, 2)
+Normal toric variety
+
+julia> b = torusinvariant_prime_divisors(B2)[1]
+Torus-invariant, prime divisor on a normal toric variety
+
+julia> w = literature_model(arxiv_id = "1208.2695", equation = "B.19", base_space = B2, model_sections = Dict("b" => b), completeness_check = false)
+Construction over concrete base may lead to singularity enhancement. Consider computing singular_loci. However, this may take time!
+
+Weierstrass model over a concrete base -- U(1) Weierstrass model based on arXiv paper 1208.2695 Eq. (B.19)
+
+julia> blow_up(w, ["x", "y", "x1"]; coordinate_name = "e1")
+Partially resolved Weierstrass model over a concrete base -- U(1) Weierstrass model based on arXiv paper 1208.2695 Eq. (B.19)
+```
+"""
+function blow_up(m::AbstractFTheoryModel, ideal_gens::Vector{String}; coordinate_name::String = "e")
+  R = cox_ring(ambient_space(m))
+  I = ideal([eval_poly(k, R) for k in ideal_gens])
+  return blow_up(m, I; coordinate_name = coordinate_name)
+end
+
+function _my_proper_transform(ring_map::T, p::MPolyRingElem, coordinate_name::String) where {T<:SetElem}
+  total_transform = ring_map(ideal([p]))
+  _e = eval_poly(coordinate_name, codomain(ring_map))
+  exceptional_ideal = total_transform + ideal([_e])
+  strict_transform, exceptional_factor = saturation_with_index(total_transform, exceptional_ideal)
+  return gens(strict_transform)[1]
+end
+
+function blow_up(m::AbstractFTheoryModel, I::MPolyIdeal; coordinate_name::String = "e")
+  
+  # Cannot (yet) blowup if this is not a Tate or Weierstrass model
+  entry_test = ((typeof(m) == GlobalTateModel) || (typeof(m) == WeierstrassModel))
+  @req entry_test "Blowups are currently only supported for Tate and Weierstrass models"
+
+  # This method only works if the model is defined over a toric variety over toric scheme
+  @req typeof(base_space(m)) <: NormalToricVariety "Blowups of Tate models are currently only supported for toric bases"
```

Comment:
General note: don't use `typeof(x) <: T`, use `x isa T`
```suggestion
  @req base_space(m) isa NormalToricVariety "Blowups of Tate models are currently only supported for toric bases"
```

Reply:
Thank you. Good suggestion. Just applied it in https://github.com/oscar-system/Oscar.jl/pull/3146/commits/9c973c1931e0694bb1d45fe48602aae69220a35d.